components/brave_wallet/browser/eth_data_parser.cc

```diff
@@ -180,7 +227,9 @@ GetTransactionInfoFromData(const std::vector<uint8_t>& data) {
         std::vector<std::string>{"bytes",     // fill path,
                                  "uint256",   // maker amount
                                  "uint256"},  // taker amount
-        std::vector<std::string>{fill_path, tx_args.at(1), tx_args.at(2)});
+        std::vector<std::string>{fill_path,
+                                 decoded_calldata.value()[1].GetString(),
+                                 decoded_calldata.value()[2].GetString()});
   } else if (selector == kSellToUniswapSelector) {
```

Comment:
why this doesn't match with `type`?

Reply:
See https://github.com/brave/brave-core/pull/23730#discussion_r1624761801